linux/Makefile

```diff
@@ -11,15 +11,57 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-TARGETS=all clean
+include engine.mk
+
+ENGINE_LIBRARY=library/libflutter_engine.so
+ENGINE_HEADER=library/include/embedder.h
+TARGETS=all
 SUBDIRS=library example
 
+$(TARGETS): $(SUBDIRS)
+$(SUBDIRS): $(ENGINE_LIBRARY) $(ENGINE_HEADER)
+	$(MAKE) -C $@ $(MAKECMDGOALS)
+
+clean: clean_local
+	rm -f $(ENGINE_LIBRARY)
+	rm -f $(ENGINE_HEADER)
+	rm -f $(FLUTTER_ENGINE_GCLIENT) $(FLUTTER_ENGINE_ROOT_DIR)/.gclient_entries
+	rm -rf $(FLUTTER_ENGINE_SYNC)
+	rm -f $(ENGINE_LIBRARY)
+	rm -f $(ENGINE_HEADER)
+
+# Only cleans this folder (library and example)
+clean_local:
```

Comment:
Is having 'clean' rm things outside this directory common? It seems potentially pretty surprising, so given that it's destructive I'd think we would want to err on the side of being conservative and have clean_all/clean rather than clean/clean_local.

Reply:
Ah, yeah I can see that being a bit of a misstep when trying to clean up things locally